docs/proposals/removal-tekton-resources.md

```diff
@@ -0,0 +1,269 @@
+
+## Proposal
+
+### User Stories
+
+- As build strategy author, I do not want to specify how sources are loaded so that I can focus on transforming sources to a container. The download of sources from potentially different systems should continue to be covered by the Shipwright Build system.
+- As build strategy author, I want to use well-documented parameters to access the path to the source code, or the name of the output image, so that I do not need to know the internal directory structure (for example `/workspace/source` should not be present in a build strategy).
+- As Shipwright Build developer, I want to continue to store metadata about the source (git commit sha) and the output (image digest) in the TaskRun status so that we can eventually expose this information for the Build user in the status of a BuildRun.
+
+### Implementation Details/Notes/Constraints [optional]
+
+We are going to use several concepts to replace Tekton's resource capability:
+
+#### 1. Tekton workspace
+
+By using resources, Tekton automatically creates sub-directories under `/workspace`, specifically, the input resource name becomes the directory name. That's why there is a `/workspace/source` directory. Outputs go into `/workspace/output`. Therefore, our image resource leads to the directory `/workspace/output/image`.
+
+With our own logic taking over responsibility on downloading sources, we also need to make sure there is a directory for this. We will create a [Tekton workspace](https://github.com/tektoncd/pipeline/blob/v0.21.0/docs/workspaces.md#using-workspaces-in-tasks) named `source` backed by an `emptyDir` volume which will then also lead to the `/workspace/source` directory.
+
+For the output image, no specific directory will be required anymore.
+
+#### 2. Tekton Parameters
+
+To pass information from the user's Build and BuildRun into the steps defined by the build strategy administrator, we will expand our usage of [Tekton parameters](https://github.com/tektoncd/pipeline/blob/v0.21.0/docs/tasks.md#specifying-parameters). A set of system parameters will be made available:
+
+TODO: which naming pattern do we want (all upper case or all lower case ? Include a `system-` prefix to prevent conflicts with parameters defined by the build strategy author?)
+
+| Name | Purpose |
+| ---- | ------- |
+| `image-url` | The URL of the image to be built (without the tag). |
+| `image-tag` | The tag of the image to be built. If the user's image URL is without a tag, then the value is `latest`. TBD: do we want this separation? Background: In the ko build strategy we currently parse the URL to extract the tag as the `ko` command wants this separately. There it would be nice to have the tag in a separate parameter, for everything else, this is not necessary. |
```

Comment:
It doesn't strike me as necessary to separate these. Maybe in the future we'll want to, but for now I think we can drop this.

Reply:
Okay. Is also my tendency as there is only one strategy atm that needs to separate it - and separating it in a script is not trivial but also not rocket-science.